docs/further_useful_features.rst

```diff
@@ -0,0 +1,459 @@
+performed if the column exists in both tables and is of the same data type: guaranteeing
+that their types are the same is part of schema validation in ``kartothek``.
+
+For example:
+
+.. ipython:: python
+
+    df.dtypes
+    different_df = pd.DataFrame(
+        {"B": pd.to_datetime(["20130102", "20190101"]), "L": [1, 4], "Q": [True, False]}
+    )
+    different_df.dtypes
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "multiple_partitioned_tables",
+        [{"data": {"table1": df, "table2": different_df}}],
+        partition_on="B",
+    )
+
+    sorted(dm.partitions.keys())
+
+
+As noted above, when data is appended to a dataset, ``kartothek`` guarantees it has
+the proper schema and partitioning.
+
+.. note:: Every partition must have data for every table. An empty dataframe in this
+          context is also considered as data.
+
+
+Secondary Indices
+-----------------
+
+The ability to build and maintain 'secondary' indices are an additional ability
+provided by ``kartothek``. In general, an index is a data structure used to improve
+the speed of read queries. In the context of ``kartothek`` an index is a data structure
+that contains a mapping of every unique value of a given column to references to all the
+partitions where this value occurs.
+
+Explicitly partitioning on a column internally creates a :class:`~kartothek.core.index.PartitionIndex`.
+While this index has a one-to-one mapping of column values to partition references,
+secondary indices have the advantage of being able to contain one-to-many mappings of
+column values to partition references.
+
+.. note::
+
+    The examples we've looked at so far have mostly used functions from the ``eager``
+    backend. As noted earlier, the ``iter`` backend executes operations on the dataset
+    on a per-partition basis and accordingly data inputs are expected to be generators.
+    Although using other iterables such as lists also works, doing so is counter
+    to the intent of the ``iter`` backend (lists would be appropriate in ``eager``).
+
+
+Writing a dataset with a secondary index:
+
+.. ipython:: python
+
+    from kartothek.io.iter import store_dataframes_as_dataset__iter
+
+    # "Generate" 5 dataframes
+    df_gen = (
+        pd.DataFrame({"date": pd.Timestamp(f"2020-01-0{i}"), "X": np.random.choice(10, 10)})
+        for i in range(1, 6)
+    )
+
+    dm = store_dataframes_as_dataset__iter(
+        df_gen,
+        store_factory,
+        "secondarily_indexed",
+        partition_on="date",
+        secondary_indices="X",
+    )
+    dm
+
+    dm = dm.load_all_indices(store_factory())
+    dm.indices["X"].eval_operator("==", 0)  # Show files where `X == 0`
+
+
+As can be seen from the example above, both ``partition_on`` and ``secondary_indices``
+can be specified together. Multiple ``secondary_indices`` can also be added as a list of
+strings.
+
+In general, secondary indices behave like explicit partitions in terms of when and how they
```

Comment:
This is a very general statement which is probably not even true.
Technically speaking you can build a secondary index for *every* fields but you cannot partition on every field (since there would be no data left after the partitioning). This is only the very first restriction which comes to my mind and there are potentially even more. Thinking about it, I believe you can also build a secondary index on a field which only exists in a single table but cannot partition on it (Don't mention this. this is not a feature, rather a (un)happy coincident atm)

I would refrain from using _In general_ since _general_ implications and similarities are hard to pin down. The actual message you want to pass to the user is: _Mind that the order of partition_on matters_ and this is something I would expect to be part of the actual `partition_on` documentation and not part of the getting started guide, i.e. please check in the `default_docs` implementation if this is the case and add it there.

Reply:
Simplified this, saying column order matters for `partition_on`, doesn't for `secondary_indices`. Note that this is part of the partitioning section already, this is not the getting started guide